### PR TITLE
162633498: Send user successful registration email

### PR DESCRIPTION
**Pull request by jeanjoe**
**What does this PR do?**

Send verification link to user email and in order to activate their account.

**Description of Tasks completed**

* Send email to user upon successful registration
* Set default is_active to False
* Set is_active to True for superuser
* add verification endpoint
* Add Tests for verification endpoints

**How to manually Test this.**
- `python manage.py makemigrations`
- `python manage.py migrate`
- To test the endpoint, Register as a normal user with a valid email address to receive the activation link.
`http://127.0.0.1:8000/api/v1/users/`

**How to run the Test**
- `python manage.py test`

Background context
The following files where modified to make the tests run successfully

- `authors/settings.py`
- `authors/apps/authentication/models.py`
- `authors/apps/authentication/urls.py`
- `authors/apps/authentication/views.py`
- `authors/apps/authentication/tests/base.py`
- `authors/apps/authentication/tests/test_jwt_auth.py`
- `authors/apps/authentication/tests/test_user_login.py`

**Relevant pivotal tracker story**

[DELIVERS #162633498]
(https://www.pivotaltracker.com/n/projects/2231917/stories/162633498)

### CODE REVIEW

File: authors/apps/core/email_handler.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+def email_template(subject, content, to):
+    #Email sending Template
+    msg = EmailMessage(
+        subject, content,
+        'no-reply@authorshaven.com',[to]
+    )
+    msg.content_subtype = 'html'
+    msg.send()
+    
+    return msg
```
Comment (mycok) [inline authors/apps/core/email_handler.py]: Add a new line

File: authors/apps/authentication/views.py [hunk 2]
```diff
@@ -25,10 +32,68 @@ def post(self, request):
...
+                'user-account-verification', args=(token, encoded_email,)
+            )
+        )
+        subject = 'Author\'s Haven Account Verification.'
+        content = '<p><strong>Hello {}!</strong> </p> \
+        <p>Thank you for registering with Authors Haven. \
+        Please follow the link below to activate your account.</p> \
+        <br> {}'.format((user.username).capitalize(), url)
+        
+        return email_template(
```
Comment (mycok) [inline authors/apps/authentication/views.py]: put all this on one line

**Comment by mubstimor** [comment]
LVGTM

**Comment by Qadriyah** [comment]
LGTM

**Comment by NabunyaLilian** [comment]
LVGTM

**Pull request merged**